platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpMetadata.java

```diff
@@ -148,11 +163,43 @@ private void initCertificates() {
     }
   }
 
-  private EntityDescriptor getEntityDescriptor() {
+  /**
+   * If the metadata is past its validity date, the SAML standard prohibits the metadata data from
+   * being used. The SAML entity is cleared if the metadata is invalid. This forces the class to
+   * attempt to retrieve the SAML entity from its source. If metadata is expired, it can continue to
+   * be used, but class attempts to get a new copy of it from the source. If successful, it clears
+   * the entity cache. If the entity cache is empty, the class attempts to retrieve it from the
+   * source.
+   *
+   * @return The root SAML entity descriptor or null
+   */
+  @Nullable
+  EntityDescriptor getEntityDescriptor() {
```

Comment:
❗️ Should be marked `synchronized` to optimize synchronizations. Verify other existing methods to see if they would benefit.

Reply:
I moved the synchronization around a bit. I had to restrict access to a couple of methods, but I was able to remove The changes remove 2 (and depending on branch evaluation, up to 4) lock/unlock synchronizations.